keps/sig-network/0006-20180314-coredns-GA-proposal.md

```diff
@@ -0,0 +1,134 @@
+---
+kep-number: TBD
+title: Switch default DNS to CoreDNS 
+authors:
+  - "@jbelamaric"
+  - "@rajansandeep"
+owning-sig: sig-network
+participating-sigs:
+  - sig-cluster-lifecycle
+reviewers:
+  - "@bowie"
+  - "@thockin"
+approvers:
+  - "@thockin"
+editor: TBD
+creation-date: 2018-03-14
+last-updated: 2018-03-14
+status: provisional
+---
+ 
+# Switch default DNS to CoreDNS 
+
+
+## Table of Contents
+
+* [Summary](#summary)
+* [Motivation](#motivation)
+    * [Goals](#goals)
+    * [Non-Goals](#non-goals)
+* [Proposal](#proposal)
+    *[User Stories](#user-stories)
+    *[Implementation Details/Notes/Constraints](#implementation-details/notes/constraints) 
+
+## Summary
+
+CoreDNS is a CNCF project dedicated to the DNS service. It has developed a K8s interface that is conform to existing kube-dns specs.
+Currently, we are following the road-map defined [here](https://github.com/kubernetes/features/issues/427). CoreDNS is Beta in Kubernetes v1.10, which can be installed as an alternate to kube-dns.
+The purpose of this proposal is to graduate CoreDNS to GA and replace the current implementation of kube-dns service by CoreDNS.
+
+## Motivation
```

Comment:
I think the things we say in "Motivation" should be things that fit with "We want to do this because..."
e.g. We want the Cluster DNS to be more flexible and extensible than what kube-dns currently offers.

IMO that is the core motivation.  Addtional benefits are, that it happens to perform better (but if it didnt we'd still be trying to convince them to move to CoreDNS, so its not really a core motivation)